pkg/security/resolvers/sbom/resolver.go

```diff
@@ -366,53 +381,63 @@ func (r *Resolver) removePendingScan(containerID containerutils.ContainerID) {
 }
 
 // analyzeWorkload generates the SBOM of the provided sbom and send it to the security agent
-func (r *Resolver) analyzeWorkload(sbom *SBOM) error {
-	sbom.Lock()
-	defer sbom.Unlock()
+func (r *Resolver) analyzeWorkload(sb *SBOM) error {
+	sb.Lock()
+	defer sb.Unlock()
 
-	seclog.Infof("analyzing sbom '%s'", sbom.ContainerID)
+	seclog.Infof("analyzing sbom '%s'", sb.ContainerID)
 
-	if sbom.state.Load() != pendingState {
-		r.removePendingScan(sbom.ContainerID)
+	if sb.state.Load() != pendingState {
+		r.removePendingScan(sb.ContainerID)
 
 		// should not append, ignore
-		seclog.Warnf("trying to analyze a sbom not in pending state for '%s': %d", sbom.ContainerID, sbom.state.Load())
+		seclog.Warnf("trying to analyze a sbom not in pending state for '%s': %d", sb.ContainerID, sb.state.Load())
 		return nil
 	}
 
 	// bail out if the workload has been analyzed while queued up
 	r.dataCacheLock.RLock()
-	if data, exists := r.dataCache.Get(sbom.workloadKey); exists {
+	if data, exists := r.dataCache.Get(sb.workloadKey); exists {
 		r.dataCacheLock.RUnlock()
-		sbom.data = data
+		sb.data = data
 
-		r.removePendingScan(sbom.ContainerID)
+		r.removePendingScan(sb.ContainerID)
 
 		return nil
 	}
 	r.dataCacheLock.RUnlock()
 
-	report, err := r.doScan(sbom)
-	if err != nil {
-		return err
+	scanTime := time.Now()
+	report, scanErr := r.doScan(sb)
+	if scanErr != nil {
+		return scanErr
+	}
+	scanDuration := time.Since(scanTime)
+
+	scanResult := &sbom.ScanResult{
+		CreatedAt: scanTime,
+		Duration:  scanDuration,
+		RequestID: string(sb.ContainerID),
+		Report:    report,
 	}
+	r.NotifyListeners(SBOMComputed, scanResult)
```

Comment:
Are we adding a test for this new functionality?

Reply:
I added a test regarding the notification mechanism.
That being said, this functionality would deserve a bigger e2e test that I suggest to put in a dedicated PR.

Reply:
Thanks. Agree on a new e2e test on separate to avoid increasing the scope of this one 😄